engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/function/FormulaFunction.java

```diff
@@ -83,7 +84,9 @@ public FormulaFunction() {
   }
 
   public Boolean getFailOnError() {
-    return failOnError;
+    return failOnError == null
```

Comment:
We use reflection to read and write properties for expressions. Hijacking the getter to never return null means the UI never sees null, and as soon as a user sets the value to null, your default catch comes in and "lies" that the value is whatever the system config says.

Likewise when saving the report: We use reflection for that as well, and call this getter. So we never write null anymore, and upon parsing the PRPT there will be a defined value here. So a loaded report will never fall back to the config. 

Work around that by introducing a new getter-only property "getComputedFailOnError" or so, that returns true or false based on your fallback logic. This way the original getter does not have to "lie" and both the expression's logic and the xml-writer and GUI-editors can go on living happily ever after.